packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -480,56 +805,77 @@ const DialogueCheckBox: React.FC<{
   );
 };
 
-const isMatched = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks.some(check => check.targetUserId === targetUserId && check.match);
-};
-
-const isChecked = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
-  return userDialogueChecks?.find(check => check.targetUserId === targetUserId)?.checked || false;
-};
-
-type MatchDialogueButtonProps = {
-  isMatched: boolean;
-  targetUserId: string;
-  targetUserDisplayName: string;
-  currentUser: UsersCurrent;
-  loadingNewDialogue: boolean;
-  createDialogue: (title: string, participants: string[]) => void;
-  classes: ClassesType;
-};
-
 const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
   isMatched,
+  checkId,
   targetUserId,
   targetUserDisplayName,
   currentUser,
-  loadingNewDialogue,
-  createDialogue,
   classes,
 }) => {
+
+  const { openDialog } = useDialog();
+  const { history } = useNavigation();
+
+  const {loading: userLoading, results} = useMulti({
+    terms: {
+      view: "dialogueMatchPreferences",
+      dialogueCheckId: checkId,
+      limit: 1000,
+    },
+    fragmentName: "DialogueMatchPreferenceInfo",
+    collectionName: "DialogueMatchPreferences",
+  });
+
   if (!isMatched) return <div></div>; // need this instead of null to keep the table columns aligned
 
-  return (
-    <div>
+  const userMatchPreferences = results && results.length > 0 ? results[0] : null;
```

Comment:
Can be simplified to:
```suggestion
  const userMatchPreferences = results?.[0]
```

Reply:
woah suggestions UI, cool